compiler-testing/src/main/kotlin/com/squareup/anvil/compiler/testing/CompilationModeTest.kt

```diff
@@ -0,0 +1,76 @@
+package com.squareup.anvil.compiler.testing
+
+import com.rickbusarow.kase.DefaultTestEnvironment
+import com.rickbusarow.kase.Kase1
+import com.rickbusarow.kase.KaseTestFactory
+import com.rickbusarow.kase.ParamTestEnvironmentFactory
+import com.rickbusarow.kase.files.HasWorkingDir
+import com.rickbusarow.kase.files.TestLocation
+import com.rickbusarow.kase.kases
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+
+public class CompilationModeTestEnvironment(
+  override val mode: CompilationMode,
+  hasWorkingDir: HasWorkingDir,
+) : DefaultTestEnvironment(hasWorkingDir),
+  CompilationEnvironment {
+
+  public companion object : ParamTestEnvironmentFactory<
+    Kase1<CompilationMode>,
+    CompilationModeTestEnvironment,
+    > {
+    override fun create(
+      params: Kase1<CompilationMode>,
+      names: List<String>,
+      location: TestLocation,
+    ): CompilationModeTestEnvironment = CompilationModeTestEnvironment(
+      mode = params.a1,
+      hasWorkingDir = HasWorkingDir(names, location),
+    )
+  }
+}
+
+public sealed interface CompilationMode {
+  public val useKapt: Boolean
+
+  public data class K2(
+    override val useKapt: Boolean,
+
+  ) : CompilationMode
+
+  @Deprecated("K1 is not implemented yet", level = DeprecationLevel.ERROR)
+  public data class K1(
```

Comment:
Is this needed? Probably relates to my other question of whether this module is intended to just be for K2 testing, but also this appears to be unused for now

Reply:
In the future, possibly.  It's intended to help in adding K2 support to all the existing tests.